modules/ui/map3d_viewer.js

```diff
@@ -251,6 +254,39 @@ export function uiMap3dViewer(context) {
           .duration(200)
           .style('opacity', '1')
           .on('end', () => redraw());
+        urlhash.setParam('map3d', 'true');
+        }
+    }
+
+    function _hashchange(){
+      // _isHidden = !_isHidden;
+      let _isHidden = context.systems.urlhash?.getParam('map3d');
+
```

Comment:
Once you have things working, refactor this method so that it is not a nearly-100% clone of the toggle method.  I get that you were just trying to get it working - an honorable goal! but in the process of getting stuff working, try to refactor so that you are writing as few new lines of code as possible. 

If it's impossible to simply call `toggle`, try to extract the common code between what `toggle` and `hashchange` both need into a single method that both can call. 

It is more work now, but your future self will thank you.

Reply:
Got rid of `_hashchange()` and now I just call refactored `toggle()` on urlchange. Looks much cleaner now, thanks again!